eZ/Publish/Core/Persistence/Legacy/Content/Location/Gateway/DoctrineDatabase.php

```diff
@@ -1330,39 +1336,49 @@ public function removeElementFromTrash($id)
     /**
      * Set section on all content objects in the subtree.
      *
-     * @param mixed $pathString
-     * @param mixed $sectionId
+     * @param string $pathString
+     * @param int $sectionId
      *
      * @return bool
      */
     public function setSectionForSubtree($pathString, $sectionId)
     {
-        $query = $this->handler->createUpdateQuery();
-
-        $subSelect = $query->subSelect();
-        $subSelect
-            ->select($this->handler->quoteColumn('contentobject_id'))
-            ->from($this->handler->quoteTable('ezcontentobject_tree'))
+        $selectContentIdsQuery = $this->connection->createQueryBuilder();
+        $selectContentIdsQuery
+            ->select($this->connection->quoteIdentifier('contentobject_id'))
+            ->from($this->connection->quoteIdentifier('ezcontentobject_tree'))
```

Comment:
you can avoid having the quoting and make it more readable by doing something like:
```php
    ->select('t.contentobject_id')
    ->from('ezcontentobject_tree', 't')
```

same below